cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryProgramRunDispatcher.java

```diff
@@ -280,6 +280,20 @@ public ProgramController dispatchProgram(ProgramRunDispatcherContext dispatcherC
     }
 
     if (!tetheredRun) {
+      if (artifactsComputeHash && (artifactsComputeHashSnapshot ||
+        !artifactDetail.getDescriptor().getArtifactId().getVersion().isSnapshot())) {
```

Comment:
nit: let's have a variable for the ArtifactId to shorten all of these lines

Reply:
Introduced artifactDescriptor to make it shorter. Since we have many artifactIDs, it might become a bit confusing for readers.